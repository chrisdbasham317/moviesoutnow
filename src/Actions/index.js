export const isLoading = (bool) => ({
  type: 'IS_LOADING',
  isLoading: bool
})

export const hasErrored = (msg) => ({
  type: 'HAS_ERRORED',
  msg
})

export const setRecentMovies = (movies) => ({
  type: 'SET_RECENT_MOVIES',
  movies
})

export const setGenres = (genres) => ({
  type: 'SET_GENRES',
  genres
})

export const currentUser = currentUser => ({
  type: 'CURRENT_USER',
  currentUser
})

export const toggleModal = toggleState => ({
  type: 'TOGGLE_MODAL',
  toggleState
})

export const signOut = () => ({
  type: 'SIGN_OUT'
})

export const setFavorites = (favorites) => ({
  type: 'SET_FAVORTIES',
  favorites
})