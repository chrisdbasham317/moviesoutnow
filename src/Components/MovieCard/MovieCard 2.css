.movieCard {
  display: flex;
  flex-direction: column;
  justify-content: flex-end;
  width: 310px;
  margin: 20px 10px;
  border: 3px solid red;
  padding: 10px;

}

.moviePoster {
  position: relative;
  width: 100%;
  height: auto;
}

.moviePosterOverlay {
  display: flex;
  flex-direction: column;
  align-items: center;
  height: 20%;
  width: 284px;
  position: absolute;
  background-color: rgba(140, 139, 139, .7);
}

.movieCardTitle,
.movieCardGenre {
  color: red;
}

movieCardGenre {
  border: 1px sold black;
  border-radius: ;
}
