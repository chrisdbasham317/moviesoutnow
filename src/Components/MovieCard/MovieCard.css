.movieCard {
  display: flex;
  flex-direction: column;
  justify-content: flex-end;
  width: 300px;
  margin: 20px 10px;
}

.moviePoster {
  position: relative;
  width: 100%;
  height: auto;
}

.moviePosterOverlay {
  display: flex;
  flex-direction: column;
  align-items: center;
  height: 20%;
  width: 300px;
  position: absolute;
  background-color: rgba(140, 139, 139, .7);
}

.movieCardTitle,
.movieCardGenre {
  color: red;
}
/* <img src="someimage.png" alt="alternative text" style="position: relative;" /><span style="position: absolute; top: 50px; left: 50px;">Some text..</span> */
