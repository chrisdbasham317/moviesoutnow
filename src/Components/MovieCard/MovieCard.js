import React from 'react';

const MovieCard = ({ title, posterPath, genre  }) => {
  return (
    <div>
      <h1>{title}</h1>
      <h2>{genre.name}</h2>
      <img src={`https://image.tmdb.org/t/p/w500${posterPath}`} alt={`${title} art`} />
    </div>
  )
}

export default MovieCard