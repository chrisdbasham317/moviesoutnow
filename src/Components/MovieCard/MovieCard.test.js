import React from 'react';
import { shallow } from 'enzyme';
import { removeFavorite, addFavorite, setCurrentMovie, clickHandler } from '../../Actions';
import { MovieCard } from './MovieCard';

describe("MovieCard", () => {
  let wrapper

  beforeEach(() => {
    wrapper = shallow(
      <MovieCard
        title={'Movie'}
        posterPath={'/path'}
        releaseDate={'yesterday'}
        voteAverage={20}
        overview={'An overview'}
        genre={'Comedy'}
        isFavorite={false}
        id={4575557}
        currentUser={{ id: 3, name: 'chris' }}
        removeFavorite={jest.fn()}
        addFavorite={jest.fn()}
        clickHandler={jest.fn()}
      />)
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  })
})