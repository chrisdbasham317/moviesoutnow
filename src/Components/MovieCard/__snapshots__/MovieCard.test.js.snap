// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieCard should match the snapshot 1`] = `
<div
  className="movieCard"
  style={
    Object {
      "border": "3px solid undefined",
    }
  }
>
  <div
    className="movie--content__div"
  >
    <img
      alt="Movie art"
      className="moviePoster"
      src="https://image.tmdb.org/t/p/w500/path"
    />
    <div
      className="moviePosterOverlay"
    >
      <div
        className="movieCardGenre__div"
      >
        <h2
          className="movieCardGenre"
          style={
            Object {
              "background": "undefined",
              "borderRadius": "18px",
              "padding": "5px 15px",
              "width": "11vw",
            }
          }
        />
        <h1
          className="movieCardTitle"
        >
          Movie
        </h1>
      </div>
      <footer
        className="footer--accents"
      >
        <div
          className="bottom-bar not-favorited"
          onClick={[Function]}
        >
          <img
            alt="is not favorite"
            className="favorite-icon"
            src="unfavorited__movie.png"
          />
          <button
            className="view--movies button--border"
            onClick={[Function]}
            type="button"
          >
            <Link
              className="view--movies"
              to="/movies/4575557"
            >
              View Movie
            </Link>
          </button>
          <h3
            className="vote--avg--text"
          >
            200%
          </h3>
        </div>
      </footer>
    </div>
  </div>
</div>
`;
